testData/pkg/apis/fun/v1alpha1/toy_types.go

```diff
@@ -79,6 +82,9 @@ type ToySpec struct {
 	// Using this for testing purpose.
 	Rook *intstr.IntOrString `json:"rook"`
 
+	// +kubebuilder:validation:AnyOf[]=type:boolean,format:bool;type:number,format:int32
```

Comment:
no, I mean having format with anything other that `type:string` is invalid openapi/jsonschema, IIRC.  we need to support it being there, but also not being there.

Reply:
@DirectXMan12  Ack. I've included your feedback in this PR. Kindly review it.